janusgraph-core/src/main/java/org/janusgraph/graphdb/internal/AbstractElement.java

```diff
@@ -67,7 +67,7 @@ public boolean equals(Object other) {
             return getCompareId()==((AbstractElement)other).getCompareId();
         } else if (other instanceof JanusGraphElement) {
             return ((JanusGraphElement) other).hasId() && getCompareId()==((JanusGraphElement)other).longId();
-        } else if (other instanceof Element) {
+        } else if (other != null) {
```

Comment:
Why this change from `if (other instanceof Element)` to `if (other != null)`?

The new check is always true, because the first line checks for `other == null`; however, the next line casts `other` to `Element`, so we really should have checked whether `other instanceof Element` is true.

Should this line be reverted to the original?

Reply:
@mbrukman At this point other is an instance of Element since otherwise it wouldn't pass the check [here](https://github.com/JanusGraph/janusgraph/blob/bfa3c970555932d4983155168322f72297aaee85/janusgraph-core/src/main/java/org/janusgraph/graphdb/internal/AbstractElement.java#L61-L63). But I did just push an update to remove the null check since you're right null was checked above.